server/src/models/database/socobo-user.ts

```diff
@@ -1,16 +1,64 @@
+import { IsEmail, IsNotEmpty, MinLength, ValidateIf } from "class-validator";
 import { ProviderType } from "./../enums/provider-type";
 import { Role } from "./../enums/role";
+import { ValidationGroup } from "./../enums/validation-group";
 
 export class SocoboUser {
+
   public id: number;
+
+  @ValidateIf((o) => o.email === "", {
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @IsNotEmpty({
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @MinLength(5, {
+    groups: [ ValidationGroup.LOGIN ]
+  })
   public username: string;
+
+  @ValidateIf((o) => o.username === "", {
```

Comment:
If a have a valid username and a invalid email the email will not be checked but is still used for authentication -> 401 (but I think it should be a 400)
Example body:
{
	"username": "test1@test.com",
	"email": "test1",
	"password": "testtest"
}

response:
"The user with the username/email test1 is not registered yet"

Maybe it would be better to check the values alway if they are present